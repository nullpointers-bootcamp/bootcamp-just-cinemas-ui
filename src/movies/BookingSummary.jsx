import React from "react";
import PropTypes from "prop-types";
import "./bookingSummary.css";

class BookingSummary extends React.Component {
  constructor() {
    super();
    this.state = {
      email: ""
    };
  }

  handleChange = e => {
    this.setState({
      email: e.target.value
    });
  };
  onConfirm = () => {
    const { seats, createTicket, showId } = this.props;
    createTicket(showId, seats, this.state.email);
  };
  renderTicketInformation = () => {
    const {
      ticketInformation: { bookingRefNumber }
    } = this.props;
    return (
      <div className="ticket-wrapper">
        <div className="booking-success">Booking successful</div>
        <div className="ticket-id-label">TICKET ID</div>
        <div className="ticket-id-value">{bookingRefNumber}</div>
      </div>
    );
  };

  render() {
    const {
      movieName,
      theatreName,
      screenName,
      seats,
      time,
      ticketInformation
    } = this.props;

    return (
      <div className="booking-summary-wrapper">
        <h1>Booking {ticketInformation ? "Confirmed" : "Summary"}</h1>
        <div className="booking-summary">
          <div className="item">
            <div className="label-item">
              Movie
              <div className="value"> {movieName} </div>
            </div>
          </div>
          <div className="item">
            <span className="label-item">
              Cinema
              <div className="value"> {theatreName} </div>
            </span>
          </div>
          <div className="item">
            <span className="label-item">
              Screen
              <div className="value"> {screenName} </div>
            </span>
          </div>
          <div className="item">
            <span className="label-item">
              Time
              <div className="value"> {time} </div>
            </span>
          </div>
        </div>
        <div className="booking-seats">
          Seats: <span className="seats-value">{seats.join(", ")}</span>
        </div>
        {ticketInformation ? (
          this.renderTicketInformation()
        ) : (
          <div>
            <div className="form-group">
              <label>Email address:</label>
              <input
                type="email"
                className="form-control"
                value={this.state.email}
                onChange={this.handleChange}
              />
            </div>
            {this.state.email ? (
              <button
                type="button"
                className="btn btn-success pull-right"
                onClick={this.onConfirm}
              >
                Confirm Booking
              </button>
            ) : null}
          </div>
        )}
      </div>
    );
  }
}

BookingSummary.defaultProps = {
  movieName: "",
  theatreName: "",
  screenName: "",
  seats: [],
  time: ""
};

BookingSummary.propTypes = {
  movieName: PropTypes.string.isRequired,
  theatreName: PropTypes.string.isRequired,
  screenName: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  seats: PropTypes.array.isRequired,
  createTicket: PropTypes.func.isRequired,
  showId: PropTypes.number.isRequired,
  ticketInformation: PropTypes.object.isRequired
};

export default BookingSummary;
