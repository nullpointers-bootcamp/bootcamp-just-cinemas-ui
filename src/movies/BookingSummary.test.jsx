import React from "react";
import renderer from "react-test-renderer";
import BookingSummary from "./BookingSummary";

it("should show the movie ticket pop ", () => {
  const tree = renderer
    .create(
      <BookingSummary
        seats={["A1", "A2"]}
        movieName={"test movie"}
        theatreName={"test"}
        screenName={"screenName"}
        time={"9:00"}
        createTicket={() => {}}
        showId={1}
        ticketInformation={{}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
