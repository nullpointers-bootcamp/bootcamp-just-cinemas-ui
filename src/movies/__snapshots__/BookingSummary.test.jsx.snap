// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should show the movie ticket pop  1`] = `
<div
  className="booking-summary-wrapper"
>
  <h1>
    Booking 
    Confirmed
  </h1>
  <div
    className="booking-summary"
  >
    <div
      className="item"
    >
      <div
        className="label-item"
      >
        Movie
        <div
          className="value"
        >
           
          test movie
           
        </div>
      </div>
    </div>
    <div
      className="item"
    >
      <span
        className="label-item"
      >
        Cinema
        <div
          className="value"
        >
           
          test
           
        </div>
      </span>
    </div>
    <div
      className="item"
    >
      <span
        className="label-item"
      >
        Screen
        <div
          className="value"
        >
           
          screenName
           
        </div>
      </span>
    </div>
    <div
      className="item"
    >
      <span
        className="label-item"
      >
        Time
        <div
          className="value"
        >
           
          9:00
           
        </div>
      </span>
    </div>
  </div>
  <div
    className="booking-seats"
  >
    Seats: 
    <span
      className="seats-value"
    >
      A1, A2
    </span>
  </div>
  <div
    className="ticket-wrapper"
  >
    <div
      className="booking-success"
    >
      Booking successful
    </div>
    <div
      className="ticket-id-label"
    >
      TICKET ID
    </div>
    <div
      className="ticket-id-value"
    />
  </div>
</div>
`;
